lib/public_key/src/public_key.erl

```diff
@@ -333,17 +370,37 @@ der_encode('PrivateKeyInfo', #'RSAPrivateKey'{} = PrivKey) ->
 der_encode('PrivateKeyInfo', {#'RSAPrivateKey'{} = PrivKey, Parameters}) ->
     der_encode('PrivateKeyInfo',
                {'PrivateKeyInfo', v1,
-                {'PrivateKeyInfo_privateKeyAlgorithm', ?'id-RSASSA-PSS', 
+                {'PrivateKeyInfo_privateKeyAlgorithm', ?'id-RSASSA-PSS',
                  {asn1_OPENTYPE, der_encode('RSASSA-PSS-params', Parameters)}},
                 der_encode('RSAPrivateKey', PrivKey), asn1_NOVALUE});
+der_encode('PrivateKeyInfo', #'ECPrivateKey'{parameters = {namedCurve, CurveOId},
+                                             privateKey = PrivKey}) when
+      CurveOId == ?'id-Ed25519' orelse
+      CurveOId == ?'id-Ed448' ->
+    der_encode('PrivateKeyInfo',
+               {'PrivateKeyInfo', v1,
+		{'PrivateKeyInfo_privateKeyAlgorithm', CurveOId, asn1_NOVALUE},
+                PrivKey, asn1_NOVALUE});
 der_encode('PrivateKeyInfo', #'ECPrivateKey'{parameters = Parameters} = PrivKey) ->
     der_encode('PrivateKeyInfo',
                {'PrivateKeyInfo', v1,
 		{'PrivateKeyInfo_privateKeyAlgorithm', ?'id-ecPublicKey',
                  {asn1_OPENTYPE, der_encode('EcpkParameters', Parameters)}},
                 der_encode('ECPrivateKey', PrivKey#'ECPrivateKey'{parameters = asn1_NOVALUE}), 
                 asn1_NOVALUE});
-der_encode(Asn1Type, Entity) when (Asn1Type == 'PrivateKeyInfo') or
+der_encode('OneAsymmetricKey', #'ECPrivateKey'{parameters = {namedCurve, CurveOId},
+                                               privateKey = PrivKey,
+                                               attributes = Attr,
+                                               publicKey = PubKey}) ->
+    der_encode('OneAsymmetricKey',
+               {'OneAsymmetricKey', 1,
```

Comment:
Why is not the record generated from the ASN.1 spec used here? It will handle the asn1_novalue automatically and assure the right format of the tuple.

Reply:
Which record are you referring to? I can not find a record type for  `OneAsymmetricKey`

Reply:
The SEQUENCE OneAssymetricKey is defined in PKCS8.asn1 , you added it in this PR. What do you do with the generates from PKCS8?

Reply:
Ok, I found it was confused by the different sets. I will write a ticket to make a general improvement of this type of thing in public_key. In an upcoming release.

Reply:
As there was other changes that creates the needed for this PR to be run for another night, I included a commit for improving usage of generated records in public_key.